src/braft/log.cpp

```diff
@@ -102,7 +103,7 @@ int Segment::create() {
     }
     LOG_IF(INFO, _fd >= 0) << "Created new segment `" << path 
                            << "' with fd=" << _fd ;
-    return _fd >= 0 ? 0 : -1;
+    return _fd >= 0 ? (old_fd > 0 ? ::close(old_fd) : 0) : -1;
```

Comment:
should use `old_fd >= 0` instead of `old_fd > 0` as `0` is a valid fd.

Reply:
OK.

Another question: how to this code change?
Use forced udpate or push an individual fixup commit?